adb_graphics/figures/skewt.py

```diff
@@ -39,258 +55,264 @@ class SkewTDiagram(gribdata.profileData):
             },
-            'grle': {
-                'color': 'orange',
-                'label': 'GRPL',
-                'marker': 'D',
-                'scale': 1.0,
-                'units': 'g/m2'
+            "grle": {
+                "color": "orange",
+                "label": "GRPL",
+                "marker": "D",
+                "scale": 1.0,
+                "units": "g/m2",
             },
-        })
+        }
 
-        profiles = self.atmo_profiles # dictionary
-        pres = profiles.get('pres').get('data')
-        temp = profiles.get('temp').get('data')
-        nlevs = len(pres)        # determine number of vertical levels
-        pres_sfc = pres[0]       # need correct surface pressure value!
+        pres = self.atmo_profiles.get("pres").get("data")
+        temp = self.atmo_profiles.get("temp").get("data")
         handles = []
-        gravity = 9.81           # m/s^2
+        gravity = 9.81  # m/s^2
 
-        lines = ['Vert. Integrated Amt\n(Resolved, Total)\n'\
-                 +'(supercool layers shaded,\nwith filled markers)']
+        lines = [
+            "Vert. Integrated Amt\n(Resolved, Total)\n"
+            "(supercool layers shaded,\nwith filled markers)"
+        ]
+        freezing_f = 32.0
 
         for mixr, settings in mixing_ratios.items():
             # Get the profile values
-            scale = settings.get('scale')
+            scale = settings.get("scale", 1.0)
             try:
-                profile = np.asarray(self.values(name=mixr)) * 1000. * scale
-            except errors.GribReadError:
-                print(f'missing {mixr} for hydrometeor plot, skipping that field.')
-                continue
-            mixr_total = 0.
-            for n in range(nlevs):
-                if n == 0:
-                    pres_layer = 2 * (pres_sfc - pres[n])  # layer depth
-                    pres_sigma = pres_sfc - pres_layer        # pressure at next sigma level
-                else:
-                    pres_layer = 2 * (pres_sigma - pres[n]) # layer depth
-                    pres_sigma = pres_sigma - pres_layer       # pressure at next sigma level
-                    mixr_total = mixr_total + pres_layer / gravity * profile[n]
-
-            # limit values to upper and lower values of lotting range
-            profile = np.where((profile > 0.) & (profile < 1.e-4), 1.e-4, profile)
-            profile = np.where((profile > 10.), 10., profile)
+                profile = self.values(name=mixr) * 1000.0 * scale
+            except (errors.NoGraphicsDefinitionForVariableError, IndexError, ValueError):
+                try:
+                    profile = self.values(name=mixr, level="uanat") * 1000.0 * scale
+                except errors.NoGraphicsDefinitionForVariableError:
+                    print(f"missing {mixr} for hydrometeor plot, skipping that field.")
+                    continue
+            mixr_total: units = 0.0
+            if profile.any():
```

Comment:
This is a fairly time-consuming procedure. Check to see if there are any non-zero values before trying to compute the vertically integrated value.